### PR TITLE
Expose offline conference peers in API

### PR DESCRIPTION
**Pull request by zugz**
(based on #1156)

---
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/1266)

### CODE REVIEW

**Comment by sudden6** [comment]
> So after save and load, the group numbers will be sequential. It's
arguable, but I think on balance this is better than preserving the
group numbers. Note that for the protocol it makes no difference - the
peer connections will go down, and so peers will update the group
numbers they have for us when we reconnect to that in our online packet.

Just to be 100% sure, this doesn't mean that after saving a running Tox instance all group numbers are invalid, right? Asking because qTox saves the *.tox on some actions but doesn't invalidate the group numbers.

**Comment by zugz** [comment]
* Thursday, 2018-11-29 at 06:35 -0800 - sudden6 <notifications@github.com>:

>> So after save and load, the group numbers will be sequential.
>
>Just to be 100% sure, this doesn't mean that after saving a running Tox 
>instance all group numbers are invalid, right? Asking because qTox 
>saves the *.tox on some actions but doesn't invalidate the group 
>numbers.

That's fine. It's only on loading the saved data, which is necessarily 
in a new tox instance, that the group numbers will change.

**Comment by sudden6** [comment]
ok, that answers my questions.

**Comment by sudden6** [comment]
___
*[toxcore/group.h, line 248 at r1](https://reviewable.io/reviews/toktok/c-toxcore/1266#-LSUlIIo7MEtIXUL2RgB:-LSiASeBCkgTR-9EpFjZ:b-1cap3w) ([raw file](https://github.com/toktok/c-toxcore/blob/4052ebfd11deea5ae5e3af131f089860386766d4/toxcore/group.h#L248)):*
<details><summary><i>Previously, zugz (zugz) wrote…</i></summary><blockquote>

Frozen only. How about "(frozen, if frozen is true) peernumber"? I hope that's clear?
</blockquote></details>

yeah, that's pretty clear

**Comment by sudden6** [comment]
___
*[toxcore/tox.c, line 1536 at r1](https://reviewable.io/reviews/toktok/c-toxcore/1266#-LSUsFZr7ZFUZznBrUbr:-LSiBA0f4UqgCGmm9H6Y:b-9i49jt) ([raw file](https://github.com/toktok/c-toxcore/blob/4052ebfd11deea5ae5e3af131f089860386766d4/toxcore/tox.c#L1536)):*
<details><summary><i>Previously, zugz (zugz) wrote…</i></summary><blockquote>

It's used elsewhere too in #1267.
</blockquote></details>

Also there it's immediately followed by a call to `del_groupchat`

**Comment by zugz** [comment]
* Sunday, 2018-12-02 at 01:12 -0800 - sudden6 <notifications@github.com>:

>___
>*[toxcore/tox.c, line 1536 at r1](https://reviewable.io/reviews/toktok/c-toxcore/1266#-LSUsFZr7ZFUZznBrUbr:-LSiBA0f4UqgCGmm9H6Y:b-9i49jt) ([raw file](https://github.com/toktok/c-toxcore/blob/4052ebfd11deea5ae5e3af131f089860386766d4/toxcore/tox.c#L1536)):*
><details><summary><i>Previously, zugz (zugz) wrote…</i></summary><blockquote>
>
>It's used elsewhere too in #1267.
></blockquote></details>
>
>Also there it's immediately followed by a call to `del_groupchat`

Yes, but to combine them you'd need to pass a bool to del_groupchat, 
which doesn't seem right. Deleting is deleting, leaving is something 
separate. Right?

**Comment by zugz** [comment]
* Sunday, 2018-12-02 at 01:31 -0800 - zugz <notifications@github.com>:

>* Sunday, 2018-12-02 at 01:12 -0800 - sudden6 <notifications@github.com>:
>
>>___
>>*[toxcore/tox.c, line 1536 at r1](https://reviewable.io/reviews/toktok/c-toxcore/1266#-LSUsFZr7ZFUZznBrUbr:-LSiBA0f4UqgCGmm9H6Y:b-9i49jt) ([raw file](https://github.com/toktok/c-toxcore/blob/4052ebfd11deea5ae5e3af131f089860386766d4/toxcore/tox.c#L1536)):*
>><details><summary><i>Previously, zugz (zugz) wrote…</i></summary><blockquote>
>>
>>It's used elsewhere too in #1267.
>></blockquote></details>
>>
>>Also there it's immediately followed by a call to `del_groupchat`
>
>Yes, but to combine them you'd need to pass a bool to del_groupchat,
>which doesn't seem right. Deleting is deleting, leaving is something
>separate. Right?

On second thoughts, you're right. Doing it all in one function does 
indeed seem neatest. I've just pushed a commit doing it.

(P.S. sorry to have broken out of reviewables... feel free to revert 
back to it)

File: toxcore/tox.api.h [hunk 1]
```diff
@@ -2321,6 +2321,70 @@ namespace conference {
 
   }
 
+  namespace offline_peer {
+
+    /**
+     * Error codes for peer info queries.
+     */
+    error for query {
```
Comment (iphydf) [inline toxcore/tox.api.h]: error for peer.query can be reused (I hope that there is a syntax for that that works).

**Pull request merged**